### ISSUE TITLE
Feature request: personal/private boards

### ISSUE THREAD

**Issue by internalfx** [mattermost/focalboard#107]
It would be helpful to be able to create boards that are not shared with all users.

**Comment by jespino** [comment]
Thanks for the feedback @internalfx. That is in our roadmap, but we are not yet there.

**Comment by it33** [comment]
We might note here that this project is for Focalboard Personal Desktop and Focalboard Personal Server and it's intended for personal use, or small teams. 

While I believe permissions is on the roadmap for the larger vision of Focalboard, certainly in editions for large organizations, I don't believe it's a feature for Focalboard Personal Desktop (since it's single user), and unclear if it should be in Focalboard Personal Server for small teams.

**Comment by internalfx** [comment]
@it33 For the personal desktop version, I could agree with you easily.

In my case I wanted to share my personal server with my wife.

Do you think she should be unable to create boards just for her use?
Should I have to share _all_ my boards with her?

**Comment by chenilim** [comment]
Hi @internalfx, appreciate the feedback. Our working assumption has been:
* Personal Desktop: Single user
* Personal Server: Collaborate within a team / family

In your scenario, do you want to share some boards with your wife, and have some boards private?
Would using Personal Server for the shared boards, and Personal Desktop for the private ones work?

One of the things we are trying to figure out is the best delivery package for each use (e.g. App Store for single user, self-host server for team). We want to improve the export / import system (e.g. export a single board) to make it easy to transfer data across instances, but I don't expect that will happen frequently - you might export a board from Desktop to Server once for example.

Hearing about specific use cases from you and others would help us better understand and map the packages to the scenarios. E.g. what are the kinds of shared boards you envision?

**Comment by internalfx** [comment]
@chenilim 

> In your scenario, do you want to share some boards with your wife, and have some boards private?
Yes, I also think it would be great if she was able to have private boards if she desired.

> Would using Personal Server for the shared boards, and Personal Desktop for the private ones work?
It would technically work, but would be inconvenient. It also has the problem of Me being unable to access "personal" boards from my phone.

**Comment by chenilim** [comment]
Going through and cleaning up old issues. The current plan to address this is to integrate with Mattermost. See the [early preview here](https://github.com/mattermost/focalboard/discussions/349). This gives us the immediate benefit of Mattermost's security and permission model.
* i.e. initially, each channel would have an associated Focalboard workspace.
* Only members of that channel can access the workspace
* This works with public and private channels, so you can specify the access controls using the channel membership

If you haven't already, please check out the [early preview](https://github.com/mattermost/focalboard/discussions/349) of how we think this could work, and let us know what you think. Thanks!

**Comment by internalfx** [comment]
@chenilim Wouldn't that make focalboard much less valuable as a standalone application?

**Comment by ZevSwe** [comment]
I would also love the idea of some kind of user access on boards. Say I have a board for personal stuff, and then one for things to do around the house with my partner, maybe one temporary one with a pack list for the vacation, then I have another one with a friend for that code project we're working on. Now I don't want my partner to see my board, because I might put presents and other stuff on it. My friend also has nothing to do with our chores around the house. So giving these users their own login with restricted access would be great!
